multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -74,62 +75,64 @@ def infer_metrics(
+            elif eval_metric_name.lower() in RETRIEVAL_METRICS:
+                # Currently only support recall as validation metric in retrieval tasks.
+                validation_metric_name = RECALL
+                return validation_metric_name, eval_metric_name
+            else:
+                warnings.warn(
+                    f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type} in matching tasks. "
+                )
 
-    if eval_metric_name is not None:
-        if problem_type != BINARY and eval_metric_name.lower() in [
-            ROC_AUC,
-            AVERAGE_PRECISION,
-            F1,
-        ]:
-            raise ValueError(f"Metric {eval_metric_name} is only supported for binary classification.")
-
-        if eval_metric_name in VALID_METRICS:
-            validation_metric_name = eval_metric_name
+        if problem_type is None:
+            eval_metric_name, validation_metric_name = MATCHING_METRICS_WITHOUT_PROBLEM_TYPE[problem_type]
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for matching tasks by default. "
+            )
             return validation_metric_name, eval_metric_name
-
-        # Currently only support recall as validation metric in retrieval tasks.
-        if is_matching and eval_metric_name in RETRIEVAL_METRICS:
-            validation_metric_name = RECALL
+        elif problem_type in MATCHING_METRICS:
+            eval_metric_name, validation_metric_name = MATCHING_METRICS[problem_type]
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for {problem_type} in matching tasks by default. "
+            )
             return validation_metric_name, eval_metric_name
+        else:
+            raise NotImplementedError(f"Problem type: {problem_type} is not yet supported for matching!")
 
-        warnings.warn(
-            f"Currently, we cannot convert the metric: {eval_metric_name} to a metric supported in torchmetrics. "
-            f"Thus, we will fall-back to use accuracy for multi-class classification problems "
-            f", ROC-AUC for binary classification problem, and RMSE for regression problems.",
-            UserWarning,
-        )
+    if eval_metric_name is not None:
+        if eval_metric_name.lower() not in problem_property.supported_evaluation_metrics:
+            warnings.warn(
+                f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type}. "
+                f"The evaluation metric is changed to {problem_property.fallback_evaluation_metric} by default."
+            )
+            if problem_property.fallback_evaluation_metric is not None:
+                eval_metric_name = problem_property.fallback_evaluation_metric
```

Comment:
What's the validation metric in this case?

Reply:
It's inferred below